packages/nimble-components/src/table/specs/table-column-placeholder-hld.md

```diff
@@ -0,0 +1,200 @@
+# Placeholders for table columns HLD
+
+## Problem Statement
+
+In some cases, an application may want to display a placeholder value in the table when a record does not have a value for a given field. The application should be able to customize that placeholder based on their needs.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Nimble issue 1538](https://github.com/ni/nimble/issues/1538)
+-   [Nimble issue 1511](https://github.com/ni/nimble/issues/1511)
+
+## Implementation / Design
+
+### High-level behavior
+
+There are two places where a value is displayed in the table that placeholders need to be considered: (1) in a cell and (2) in a group row. The cell placeholder will be configurable on each column that supports having a placeholder, and it will default to an empty string. The group row placeholder will come from the table's localization provider, and it will not be configurable through a column's API.
+
+Placeholders within a cell will be rendered with nimble's placeholder font, which is currently 60% opacity of nimble's body font.
+
+Placeholders within a group row will have no special visual treatment.
+
```

Comment:
Some questions that I think I know the answer to but would be good to make explicit:
1. Will placeholders follow the same truncation behavior as cells and headers (... plus title)?
2. Will placeholders affect sort order?